contents/docs/plugins/build/index.md

```diff
@@ -4,37 +4,53 @@ sidebar: Docs
 showTitle: true
 ---
 
-A great way to understand what you can do with [plugins](/docs/plugins/overview) is to understand how data flows throw plugins.
+PostHog makes it possible to build your own [plugins](/docs/plugins/overview) and integrate with other platforms. So, if [our integration library](/integrations) is missing something you need then you may still be able to create it yourself.
 
-There are 2 big concepts to remember:
+Plugins can add more information to the event, modify existing properties or trigger a range of other activities. There are also some plugins that enqueue jobs to run in the future. Find out more about jobs in [our developer reference docs](/docs/plugins/build/reference#jobs-1).
 
-1. Every plugin acts on a single event coming in.
+Before building your first plugin it's important to understand how data flows through plugins in the first place. There are three critical concepts to remember:
 
-2. Plugins act on events before they are stored.
+1. Plugins act on _single events_ coming in to PostHog.
 
-Each plugin can add more information to the event, or even modify existing properties. For example, the [GeoIP plugin](https://posthog.com/plugins/geoip) is a plugin that adds GeoIP information to the event.
+2. Plugins act on events _before_ they are stored.
 
-Note that the output of one plugin goes into the next plugin. Here's how this looks:
+3. The output of one plugin will go into the next plugin, creating a chain.
+
+Before we get started, lets look at an examples of these principles in action. 
+
+## Example of a plugin chain
+
+The [GeoIP plugin](/integrations/geoip) is an example of a plugin which adds information to events. Specifically, it adds geographical information based on the user IP address. As above, it is triggered on each single event and adds additional informational to each event before it is stored.
+
+By running a second plugin after the GeoIP plugin, we create a plugin chain. Here's an example of how this can look for an individual event when a second plugin (which simply adds ```Hello: "world"``` to the event) runs after the GeoIP plugin. 
 
 ![GeoIP Plugin Example](../../../images/plugins/geoip-plugin-example.png)
 
-However, this isn't all. Plugins _don't_ have to modify events at all: they can do other things when an event comes in.
+Plugin chains are important because they control what the event looks like before it is stored. If you want to remove certain properties out of an event with the [Property Filter plugin](/integrations/property-filter), for example, it is best to have it run at the end of the plugin chain. 
```

Comment:
I don't get this, why should this run at the end of the chain?

Reply:
Because otherwise some other plugin might add the properties "back" (e.g. property flattener, geoip)

Reply:
Hmm, they can only come back if there's a second source of the data.. but yeah, I see how this can be a good rule of thumb.

Reply:
Yep, as Karl said was my intention. Resolving this. Thank you!